src/wp-includes/functions.wp-scripts.php

```diff
@@ -340,24 +341,28 @@ function wp_deregister_script( $handle ) {
  *                                    as a query string for cache busting purposes. If version is set to false, a version
  *                                    number is automatically added equal to current installed WordPress version.
  *                                    If set to null, no version is added.
+ * @param array            $args      {
+ *      Optional. An array of additional script strategies. Default empty array.
+ *
+ *      @type boolean   $in_footer    Optional. Default true.
+ *      @type string    $strategy     Optional. Values blocking|defer|async. Default 'blocking'.
+ * }
+ *
  * @param bool             $in_footer Optional. Whether to enqueue the script before `</body>` instead of in the `<head>`.
  *                                    Default 'false'.
  */
-function wp_enqueue_script( $handle, $src = '', $deps = array(), $ver = false, $in_footer = false ) {
+function wp_enqueue_script( $handle, $src = '', $deps = array(), $ver = false, $args = array() ) {
 	_wp_scripts_maybe_doing_it_wrong( __FUNCTION__, $handle );
 
 	$wp_scripts = wp_scripts();
 
-	if ( $src || $in_footer ) {
+	if ( $src || ! empty( $args ) ) {
 		$_handle = explode( '?', $handle );
-
 		if ( $src ) {
 			$wp_scripts->add( $_handle[0], $src, $deps, $ver );
 		}
 
-		if ( $in_footer ) {
-			$wp_scripts->add_data( $_handle[0], 'group', 1 );
-		}
+		$wp_scripts->add_data( $_handle[0], 'script_args', $args );
```

Comment:
Are there any potential side effects that could happen here now that we're always calling the `add_data` method rather than only doing so if `$in_footer` is `true`? In other words, should we still check to ensure `$in_footer` is not empty before running this method?

Reply:
Looking more deeply into this, it looks like the main difference is that previously, here and in `wp_register_script` above, the `add_data` method is only called whenever the value for `$in_footer` is not the default value. Now, we're calling this method every time these functions are called and adding the default args as data on the dependency, as well as adding `group` data whenever `$in_footer` or `$args['in_footer']` are `true`.

Is there a reason to add the default args as `script_args` data to every script, or could we bail early if the default value is set, like before? Like, this:

```suggestion
		if ( ! empty( $args ) ) {
			$wp_scripts->add_data( $_handle[0], 'script_args', $args );
		}
```

Reply:
@joemcgill on the above point, if we are normalizing data, would we perhaps not want to store those normalized args for all handles, especially when it comes time to getting eligible strategies?

Reply:
@10upsimon, I think we will want to consider the performance impact of running these methods for every handle versus only for those that are not using the default arguments. For those that _are_ using the defaults, I think we can infer the default data when determining eligible strategies.